sema4ai/src/sema4ai_code/workspace_manager.py

```diff
@@ -1,6 +1,9 @@
 import os
+from collections import defaultdict
+from collections.abc import Iterator
 from pathlib import Path
-from typing import Dict, List, Optional
+from pprint import pformat
```

Comment:
Minor: make `pformat` a local import (although maybe using `json.dumps()` could be done too as `json` is imported all over the place already).